packages/interactivity/src/utils.ts

```diff
@@ -49,9 +49,19 @@ const afterNextFrame = ( callback: () => void ) => {
  * @return Promise
  */
 export const splitTask = () => {
-	return new Promise( ( resolve ) => {
-		// TODO: Use scheduler.yield() when available.
-		setTimeout( resolve, 0 );
+	return new Promise( async ( resolve ) => {
+		if (
+			'scheduler' in window &&
+			typeof window.scheduler === 'object' &&
+			null !== window.scheduler &&
+			'yield' in window.scheduler &&
+			typeof window.scheduler.yield === 'function'
+		) {
+			await window.scheduler.yield();
+			resolve( undefined );
+		} else {
+			setTimeout( resolve, 0 );
+		}
 	} );
```

Comment:
This is more readable IMO and avoids an async function in the promise body:

```js
export const splitTask = () => {
	if (
		'scheduler' in window &&
		typeof window.scheduler === 'object' &&
		null !== window.scheduler &&
		'yield' in window.scheduler &&
		typeof window.scheduler.yield === 'function'
	) {
		return window.scheduler.yield();
	}
	
	return new Promise( ( resolve ) => {
		setTimeout( resolve, 0 );
	} );
}
```

Reply:
I've applied that in 70077b8 and then further improved it in d5c51a6 by only doing the check for `scheduler.yield()` once.

Reply:
And improved yet further in 2516cd4 by removing the wrapper function around `scheduler.yield()`. So when `scheduler.yield()` is available, `splitTask` is just a reference to that function.

Reply:
Humm. Actually this results in an error for some reason:

![image](https://github.com/user-attachments/assets/d27c9dbd-0832-4803-b7e6-0af6a3022e7f)

I guess it's because `yield` here doesn't have `scheduler` as its context object.

Reply:
Fixed in a0933f4a6dd via `bind()`.

Reply:
There's probably zero difference in the end between:

```js
window.scheduler.yield.bind( window.scheduler )
```

And:

```js
() => {
    return window.scheduler.yield();
}
```

Since in both cases a new function is involved. The former seems to be syntactic sugar for the latter.

Reply:
But why use `bind` over the latter? I find the latter more intuitive to understand. Also, I'm not familiar enough with `bind` to be confident so that _is_ the same. Since you also say "probably", I'd suggest we go with the latter.

We have to use a function anyway for the fallback of returning a promise, so we might as well wrap the entire code in a function.

Reply:
"bind" is very very slightly preferable for the JS engine, since it can represent the result as a "bound function" that forwards directly to the function it binds already while resolving the call, and doesn't need to parse it, compile it, interpret bytecode, etc.